opensrp-chw/src/miecd/java/org/smartregister/chw/interactor/PncHomeVisitInteractorFlv.java

```diff
@@ -0,0 +1,953 @@
+                );
+
+                if (StringUtils.isNotBlank(fp_method)) {
+                    String method = "";
+                    switch (fp_method) {
+                        case "None":
+                            method = context.getString(R.string.none);
+                            break;
+                        case "PPIUCD":
+                            method = context.getString(R.string.ppiucd);
+                            break;
+                        case "Pills":
+                            method = context.getString(R.string.pills);
+                            break;
+                        case "Implant":
+                            method = context.getString(R.string.implant);
+                            break;
+                        case "Condoms":
+                            method = context.getString(R.string.condoms);
+                            break;
+                        case "LAM":
+                            method = context.getString(R.string.lam);
+                            break;
+                        case "Bead Counting":
+                            method = context.getString(R.string.standard_day_method);
+                            break;
+                        case "Permanent (BTL)":
+                            method = context.getString(R.string.permanent_blt);
+                            break;
+                        case "Permanent (Vasectomy)":
+                            method = context.getString(R.string.permanent_vasectomy);
+                            break;
+                        default:
+                            break;
+                    }
+                    builder.append(MessageFormat.format("{0}: {1}", method, StringUtils.isNoneBlank(fp_start_date) ? new SimpleDateFormat("dd MMM yyyy", Locale.getDefault()).format(start_date) : ""));
+                }
+                return builder.toString();
+            }
+
+            @Override
+            public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isBlank(fp_counseling)) {
+                    return BaseAncHomeVisitAction.Status.PENDING;
+                }
+                if ("Yes".equalsIgnoreCase(fp_counseling)) {
+                    return BaseAncHomeVisitAction.Status.COMPLETED;
+                } else {
+                    return BaseAncHomeVisitAction.Status.PARTIALLY_COMPLETED;
+                }
+            }
+        };
+
+        BaseAncHomeVisitAction action = new BaseAncHomeVisitAction.Builder(context, context.getString(R.string.pnc_family_planning))
+                .withOptional(false)
+                .withDetails(details)
+                .withFormName(Constants.JSON_FORM.PNC_HOME_VISIT.getFamilyPlanning())
+                .withHelper(helper)
+                .build();
+        actionList.put(context.getString(R.string.pnc_family_planning), action);
+    }
+
+    private void evaluateExclusiveBreastFeeding() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'evaluateExclusiveBreastFeeding()' has an NPath complexity of 240](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5426631)